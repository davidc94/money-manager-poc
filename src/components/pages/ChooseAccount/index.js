import React, { PureComponent } from 'react';
import { View, TouchableOpacity, Text, ScrollView, Image, FlatList} from 'react-native';
import StyledView from '../../atoms/StyledView';
import StyledText from '../../atoms/StyledText';
import StyledButton from '../../atoms/StyledButton';

export default class ChooseAccount extends PureComponent {
  state = {
    selectAccount: false,
    data:[{
      'imgUrl' :'https://i.imgur.com/K3KJ3w4h.jpg',
      'bankName': 'HSBC Bank (UK)',
      'accountType':'Advance Account',
      'sortCode':'20-01-89',
      'accountNumber': '54923765'
    },
    {
      'imgUrl' :'https://i.imgur.com/K3KJ3w4h.jpg',
      'bankName': 'HSBC Bank (UK)',
      'accountType':'Premier Account',
      'sortCode':'20-01-89',
      'accountNumber': '54923766'
    },
    {
      'imgUrl' :'https://i.imgur.com/K3KJ3w4h.jpg',
      'bankName': 'HSBC Bank (UK)',
      'accountType':'`Corporate` Account',
      'sortCode':'20-01-89',
      'accountNumber': '54923767'
    }]
  }

  selectAccount =(account) => 
    this.setState({
      selectAccount:account
    });
  

renderItem = ({item}) => {
  const bdColor = item.accountNumber === this.state.selectAccount ? 'blue': 'lightgray';
  const bgColor = item.accountNumber === this.state.selectAccount ? '#ccebff': '#fff';
    return( 
      <TouchableOpacity onPress={ () => this.selectAccount(item.accountNumber)}>
        <View style={{flex:1, flexDirection:'row',justifyContent:"center", alignItems:"center", padding: 20, margin:10,borderColor: bdColor, backgroundColor:bgColor, borderStyle:"solid", borderWidth:1
            }}>
          <Image source={{ uri: item.imgUrl }} style={{width: 50, height: 50, flex: 1, marginRight: 30}}></Image>
          <View style={{flex:3,flexDirection:'column'}}>
            <Text style={{marginBottom:10}}>
              {item.bankName}
              </Text>
              <Text style={{marginBottom:10}}>
              {item.accountType}
              </Text>
              <Text style={{marginBottom:10}}>
              {item.sortCode}  {item.accountNumber}
            </Text>
          </View>
        </View>
     </TouchableOpacity>
    )
  }

  
  render() {
    const { navigation } = this.props;
    const thumbnail_image= 'https://i.imgur.com/K3KJ3w4h.jpg';
    return (
      <ScrollView>
        <StyledView>
            <StyledText>
              We found three accounts in your name.
              Which account do you use on a daily 
              basis?
            </StyledText>
              <FlatList
                data= {this.state.data}
                renderItem = { this.renderItem }
                extraData= {this.state}
              /> 
        </StyledView>
        <StyledView style={{ padding: 10 }}>
              <StyledButton  disabled={!this.state.selectAccount}  style={{fontSize:12, width:"100%" }}
                onPress={() => this.props.navigation.navigate('Dashboard')}>
                Select & Continue
              </StyledButton>
        </StyledView>      
       </ScrollView>  
    );
  }
}