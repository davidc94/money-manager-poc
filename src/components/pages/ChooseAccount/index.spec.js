import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ChooseAccount from './index';

configure({ adapter: new Adapter() });

describe('choose account', () => {
  const render = shallow(<ChooseAccount />);

  it('renders as expected', () => {
    expect(render).toMatchSnapshot();
  });
});
