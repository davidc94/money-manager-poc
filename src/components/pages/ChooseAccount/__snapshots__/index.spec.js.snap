// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`choose account renders as expected 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ChooseAccount
    navigation={
      Object {
        "navigate": [MockFunction],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
<<<<<<< HEAD
      "children": <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Component>
          ChooseAccount
        </Component>
      </TouchableOpacity>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <Component>
          ChooseAccount
        </Component>,
        "onPress": [Function],
=======
      "children": Array [
        <Styled(Component)>
          <StyledText>
            We found three accounts in your name. Which account do you use on a daily basis?
          </StyledText>
          <FlatList
            data={
              Array [
                Object {
                  "accountNumber": "54923765",
                  "accountType": "Advance Account",
                  "bankName": "HSBC Bank (UK)",
                  "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                  "sortCode": "20-01-89",
                },
                Object {
                  "accountNumber": "54923766",
                  "accountType": "Premier Account",
                  "bankName": "HSBC Bank (UK)",
                  "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                  "sortCode": "20-01-89",
                },
                Object {
                  "accountNumber": "54923767",
                  "accountType": "\`Corporate\` Account",
                  "bankName": "HSBC Bank (UK)",
                  "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                  "sortCode": "20-01-89",
                },
              ]
            }
            disableVirtualization={false}
            extraData={
              Object {
                "data": Array [
                  Object {
                    "accountNumber": "54923765",
                    "accountType": "Advance Account",
                    "bankName": "HSBC Bank (UK)",
                    "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                    "sortCode": "20-01-89",
                  },
                  Object {
                    "accountNumber": "54923766",
                    "accountType": "Premier Account",
                    "bankName": "HSBC Bank (UK)",
                    "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                    "sortCode": "20-01-89",
                  },
                  Object {
                    "accountNumber": "54923767",
                    "accountType": "\`Corporate\` Account",
                    "bankName": "HSBC Bank (UK)",
                    "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                    "sortCode": "20-01-89",
                  },
                ],
                "selectAccount": false,
              }
            }
            horizontal={false}
            initialNumToRender={10}
            keyExtractor={[Function]}
            maxToRenderPerBatch={10}
            numColumns={1}
            onEndReachedThreshold={2}
            renderItem={[Function]}
            scrollEventThrottle={50}
            updateCellsBatchingPeriod={50}
            windowSize={21}
          />
        </Styled(Component)>,
        <Styled(Component)
          style={
            Object {
              "padding": 10,
            }
          }
        >
          <StyledButton
            disabled={true}
            onPress={[Function]}
            style={
              Object {
                "fontSize": 12,
                "width": "100%",
              }
            }
          >
            Select & Continue
          </StyledButton>
        </Styled(Component)>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <StyledText>
              We found three accounts in your name. Which account do you use on a daily basis?
            </StyledText>,
            <FlatList
              data={
                Array [
                  Object {
                    "accountNumber": "54923765",
                    "accountType": "Advance Account",
                    "bankName": "HSBC Bank (UK)",
                    "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                    "sortCode": "20-01-89",
                  },
                  Object {
                    "accountNumber": "54923766",
                    "accountType": "Premier Account",
                    "bankName": "HSBC Bank (UK)",
                    "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                    "sortCode": "20-01-89",
                  },
                  Object {
                    "accountNumber": "54923767",
                    "accountType": "\`Corporate\` Account",
                    "bankName": "HSBC Bank (UK)",
                    "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                    "sortCode": "20-01-89",
                  },
                ]
              }
              disableVirtualization={false}
              extraData={
                Object {
                  "data": Array [
                    Object {
                      "accountNumber": "54923765",
                      "accountType": "Advance Account",
                      "bankName": "HSBC Bank (UK)",
                      "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                      "sortCode": "20-01-89",
                    },
                    Object {
                      "accountNumber": "54923766",
                      "accountType": "Premier Account",
                      "bankName": "HSBC Bank (UK)",
                      "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                      "sortCode": "20-01-89",
                    },
                    Object {
                      "accountNumber": "54923767",
                      "accountType": "\`Corporate\` Account",
                      "bankName": "HSBC Bank (UK)",
                      "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                      "sortCode": "20-01-89",
                    },
                  ],
                  "selectAccount": false,
                }
              }
              horizontal={false}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              numColumns={1}
              onEndReachedThreshold={2}
              renderItem={[Function]}
              scrollEventThrottle={50}
              updateCellsBatchingPeriod={50}
              windowSize={21}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "We found three accounts in your name. Which account do you use on a daily basis?",
            },
            "ref": null,
            "rendered": "We found three accounts in your name. Which account do you use on a daily basis?",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "data": Array [
                Object {
                  "accountNumber": "54923765",
                  "accountType": "Advance Account",
                  "bankName": "HSBC Bank (UK)",
                  "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                  "sortCode": "20-01-89",
                },
                Object {
                  "accountNumber": "54923766",
                  "accountType": "Premier Account",
                  "bankName": "HSBC Bank (UK)",
                  "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                  "sortCode": "20-01-89",
                },
                Object {
                  "accountNumber": "54923767",
                  "accountType": "\`Corporate\` Account",
                  "bankName": "HSBC Bank (UK)",
                  "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                  "sortCode": "20-01-89",
                },
              ],
              "disableVirtualization": false,
              "extraData": Object {
                "data": Array [
                  Object {
                    "accountNumber": "54923765",
                    "accountType": "Advance Account",
                    "bankName": "HSBC Bank (UK)",
                    "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                    "sortCode": "20-01-89",
                  },
                  Object {
                    "accountNumber": "54923766",
                    "accountType": "Premier Account",
                    "bankName": "HSBC Bank (UK)",
                    "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                    "sortCode": "20-01-89",
                  },
                  Object {
                    "accountNumber": "54923767",
                    "accountType": "\`Corporate\` Account",
                    "bankName": "HSBC Bank (UK)",
                    "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                    "sortCode": "20-01-89",
                  },
                ],
                "selectAccount": false,
              },
              "horizontal": false,
              "initialNumToRender": 10,
              "keyExtractor": [Function],
              "maxToRenderPerBatch": 10,
              "numColumns": 1,
              "onEndReachedThreshold": 2,
              "renderItem": [Function],
              "scrollEventThrottle": 50,
              "updateCellsBatchingPeriod": 50,
              "windowSize": 21,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
>>>>>>> feat(select-bank-account-page): add functionality
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
<<<<<<< HEAD
          "children": "ChooseAccount",
=======
          "children": <StyledButton
            disabled={true}
            onPress={[Function]}
            style={
              Object {
                "fontSize": 12,
                "width": "100%",
              }
            }
          >
            Select & Continue
          </StyledButton>,
          "style": Object {
            "padding": 10,
          },
>>>>>>> feat(select-bank-account-page): add functionality
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Select & Continue",
            "disabled": true,
            "onPress": [Function],
            "style": Object {
              "fontSize": 12,
              "width": "100%",
            },
          },
          "ref": null,
          "rendered": "Select & Continue",
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [MockFunction],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
<<<<<<< HEAD
        "children": <TouchableOpacity
          activeOpacity={0.2}
          onPress={[Function]}
        >
          <Component>
            ChooseAccount
          </Component>
        </TouchableOpacity>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "activeOpacity": 0.2,
          "children": <Component>
            ChooseAccount
          </Component>,
          "onPress": [Function],
=======
        "children": Array [
          <Styled(Component)>
            <StyledText>
              We found three accounts in your name. Which account do you use on a daily basis?
            </StyledText>
            <FlatList
              data={
                Array [
                  Object {
                    "accountNumber": "54923765",
                    "accountType": "Advance Account",
                    "bankName": "HSBC Bank (UK)",
                    "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                    "sortCode": "20-01-89",
                  },
                  Object {
                    "accountNumber": "54923766",
                    "accountType": "Premier Account",
                    "bankName": "HSBC Bank (UK)",
                    "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                    "sortCode": "20-01-89",
                  },
                  Object {
                    "accountNumber": "54923767",
                    "accountType": "\`Corporate\` Account",
                    "bankName": "HSBC Bank (UK)",
                    "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                    "sortCode": "20-01-89",
                  },
                ]
              }
              disableVirtualization={false}
              extraData={
                Object {
                  "data": Array [
                    Object {
                      "accountNumber": "54923765",
                      "accountType": "Advance Account",
                      "bankName": "HSBC Bank (UK)",
                      "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                      "sortCode": "20-01-89",
                    },
                    Object {
                      "accountNumber": "54923766",
                      "accountType": "Premier Account",
                      "bankName": "HSBC Bank (UK)",
                      "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                      "sortCode": "20-01-89",
                    },
                    Object {
                      "accountNumber": "54923767",
                      "accountType": "\`Corporate\` Account",
                      "bankName": "HSBC Bank (UK)",
                      "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                      "sortCode": "20-01-89",
                    },
                  ],
                  "selectAccount": false,
                }
              }
              horizontal={false}
              initialNumToRender={10}
              keyExtractor={[Function]}
              maxToRenderPerBatch={10}
              numColumns={1}
              onEndReachedThreshold={2}
              renderItem={[Function]}
              scrollEventThrottle={50}
              updateCellsBatchingPeriod={50}
              windowSize={21}
            />
          </Styled(Component)>,
          <Styled(Component)
            style={
              Object {
                "padding": 10,
              }
            }
          >
            <StyledButton
              disabled={true}
              onPress={[Function]}
              style={
                Object {
                  "fontSize": 12,
                  "width": "100%",
                }
              }
            >
              Select & Continue
            </StyledButton>
          </Styled(Component)>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <StyledText>
                We found three accounts in your name. Which account do you use on a daily basis?
              </StyledText>,
              <FlatList
                data={
                  Array [
                    Object {
                      "accountNumber": "54923765",
                      "accountType": "Advance Account",
                      "bankName": "HSBC Bank (UK)",
                      "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                      "sortCode": "20-01-89",
                    },
                    Object {
                      "accountNumber": "54923766",
                      "accountType": "Premier Account",
                      "bankName": "HSBC Bank (UK)",
                      "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                      "sortCode": "20-01-89",
                    },
                    Object {
                      "accountNumber": "54923767",
                      "accountType": "\`Corporate\` Account",
                      "bankName": "HSBC Bank (UK)",
                      "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                      "sortCode": "20-01-89",
                    },
                  ]
                }
                disableVirtualization={false}
                extraData={
                  Object {
                    "data": Array [
                      Object {
                        "accountNumber": "54923765",
                        "accountType": "Advance Account",
                        "bankName": "HSBC Bank (UK)",
                        "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                        "sortCode": "20-01-89",
                      },
                      Object {
                        "accountNumber": "54923766",
                        "accountType": "Premier Account",
                        "bankName": "HSBC Bank (UK)",
                        "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                        "sortCode": "20-01-89",
                      },
                      Object {
                        "accountNumber": "54923767",
                        "accountType": "\`Corporate\` Account",
                        "bankName": "HSBC Bank (UK)",
                        "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                        "sortCode": "20-01-89",
                      },
                    ],
                    "selectAccount": false,
                  }
                }
                horizontal={false}
                initialNumToRender={10}
                keyExtractor={[Function]}
                maxToRenderPerBatch={10}
                numColumns={1}
                onEndReachedThreshold={2}
                renderItem={[Function]}
                scrollEventThrottle={50}
                updateCellsBatchingPeriod={50}
                windowSize={21}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "We found three accounts in your name. Which account do you use on a daily basis?",
              },
              "ref": null,
              "rendered": "We found three accounts in your name. Which account do you use on a daily basis?",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "data": Array [
                  Object {
                    "accountNumber": "54923765",
                    "accountType": "Advance Account",
                    "bankName": "HSBC Bank (UK)",
                    "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                    "sortCode": "20-01-89",
                  },
                  Object {
                    "accountNumber": "54923766",
                    "accountType": "Premier Account",
                    "bankName": "HSBC Bank (UK)",
                    "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                    "sortCode": "20-01-89",
                  },
                  Object {
                    "accountNumber": "54923767",
                    "accountType": "\`Corporate\` Account",
                    "bankName": "HSBC Bank (UK)",
                    "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                    "sortCode": "20-01-89",
                  },
                ],
                "disableVirtualization": false,
                "extraData": Object {
                  "data": Array [
                    Object {
                      "accountNumber": "54923765",
                      "accountType": "Advance Account",
                      "bankName": "HSBC Bank (UK)",
                      "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                      "sortCode": "20-01-89",
                    },
                    Object {
                      "accountNumber": "54923766",
                      "accountType": "Premier Account",
                      "bankName": "HSBC Bank (UK)",
                      "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                      "sortCode": "20-01-89",
                    },
                    Object {
                      "accountNumber": "54923767",
                      "accountType": "\`Corporate\` Account",
                      "bankName": "HSBC Bank (UK)",
                      "imgUrl": "https://i.imgur.com/K3KJ3w4h.jpg",
                      "sortCode": "20-01-89",
                    },
                  ],
                  "selectAccount": false,
                },
                "horizontal": false,
                "initialNumToRender": 10,
                "keyExtractor": [Function],
                "maxToRenderPerBatch": 10,
                "numColumns": 1,
                "onEndReachedThreshold": 2,
                "renderItem": [Function],
                "scrollEventThrottle": 50,
                "updateCellsBatchingPeriod": 50,
                "windowSize": 21,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
>>>>>>> feat(select-bank-account-page): add functionality
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
<<<<<<< HEAD
            "children": "ChooseAccount",
=======
            "children": <StyledButton
              disabled={true}
              onPress={[Function]}
              style={
                Object {
                  "fontSize": 12,
                  "width": "100%",
                }
              }
            >
              Select & Continue
            </StyledButton>,
            "style": Object {
              "padding": 10,
            },
>>>>>>> feat(select-bank-account-page): add functionality
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Select & Continue",
              "disabled": true,
              "onPress": [Function],
              "style": Object {
                "fontSize": 12,
                "width": "100%",
              },
            },
            "ref": null,
            "rendered": "Select & Continue",
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [MockFunction],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
